src/android/Capture.java

```diff
@@ -234,7 +234,7 @@ private JSONObject getAudioVideoData(String filePath, JSONObject obj, boolean vi
         return obj;
     }
 
-    private boolean isMissingPermissions(Request req, ArrayList<String> permissions) {
+    private boolean isMissingPermissions(Request req, List<String> permissions) {
```

Comment:
I think the `ArrayList` is "expected" here, as an ordered collection, and so should not be replaced by `List`.

Reply:
Aren't every list in JAVA ordered? In the sense that the iteration is always deterministic.
Here, you give a list that is ordered as you wish it to be, we don't need to constrain to use a particular implementation of `List` : the reponsability of the order is for the caller.

In our very case, we use `Arrays.asList` that keeps the same order as the initial array.

Reply:
You're right!
To update and clean this up a little more, maybe also do?
- `ArrayList<String> missingPermissions = …` ⇒ `List<String> missingPermissions = …`
  (+ a bonus space before `:` in `String permission: permissions` 😇)
- `ArrayList<String> cameraPermissions = …` ⇒ `List<String> cameraPermissions = …`

Reply:
Good catch, indeed! Just commited that.